src/scripts/main.js

```diff
@@ -1,3 +1,298 @@
 'use strict';
 
-// write code here
+const MIN_AGE = 18;
+const MAX_AGE = 100;
+const MIN_SALARY = 50000;
+const TIMER_DURATION = 10000; // Milisecond (s = 1000ms)
+const pageBody = document.querySelector('body');
+const tableHeader = document.querySelector('thead');
+const tableHeaderCells = tableHeader.querySelector('tr');
+const tableBody = document.querySelector('tbody');
+const toNumber = (string) => {
+  if (typeof string === 'string' && string) {
+    const clearNum = string.replace(/[$, ]/g, '');
+
+    return parseInt(clearNum, 10);
+  }
+
+  return NaN;
+};
+const sortTable = (elemA, elemB) => {
+  const aToNum = toNumber(elemA);
+  const bToNum = toNumber(elemB);
+
+  if (Number.isNaN(aToNum)) {
+    return isSecondClick
+      ? elemB.localeCompare(elemA)
+      : elemA.localeCompare(elemB);
+  }
+
+  return isSecondClick ? bToNum - aToNum : aToNum - bToNum;
+};
+const cityOptions = [
+  `Tokyo`,
+  `Singapore`,
+  `London`,
+  `New York`,
+  `Edinburgh`,
+  `San Francisco`,
+];
+let clickedSortTable;
+let isSecondClick = false;
+
+const createElement = (cellName, type) => {
+  const label = document.createElement('label');
+  const input = document.createElement('input');
+
+  label.textContent = `${cellName}:`;
+  input.setAttribute('data-qa', cellName.toLowerCase());
+  input.required = true;
+  input.type = type;
+  label.appendChild(input);
+
+  return label;
+};
+
+const getElInput = (element) => {
+  return element.firstElementChild;
+};
+
+const formContainer = document.createElement('form');
+const nameElement = createElement('Name', 'text');
+const positionElement = createElement('Position', 'text');
+const officeLabel = document.createElement('label');
+const officeChoose = document.createElement('select');
+const ageElement = createElement('Age', 'number');
+const salaryElement = createElement('Salary', 'number');
+const formButton = document.createElement('button');
+const inputs = [
+  getElInput(nameElement),
+  getElInput(positionElement),
+  officeChoose,
+  getElInput(ageElement),
+  getElInput(salaryElement),
+];
+
+formContainer.classList.add('new-employee-form');
```

Comment:
The logic for creating the form should be within the function body, as this will make the code more meaningful. If the function is properly named and focuses on a single task, reading the code will be like reading a book.